TileStache/Mapnik.py

```diff
@@ -74,19 +74,24 @@ class ImageProvider:
         - http://trac.mapnik.org/wiki/XMLConfigReference
     """
 
-    def __init__(self, layer, mapfile, fonts=None, scale_factor=None):
-        """ Initialize Mapnik provider with layer and mapfile.
+    def __init__(self, layer, mapconfig, fonts=None, scale_factor=None):
+        """ Initialize Mapnik provider with layer and mapconfig.
 
-            XML mapfile keyword arg comes from TileStache config,
+            XML mapconfig keyword arg comes from TileStache config,
             and is an absolute path by the time it gets here.
         """
-        maphref = urljoin(layer.config.dirpath, mapfile)
+        maphref = urljoin(layer.config.dirpath, mapconfig)
```

Comment:
Not sure, but first, have you seen this?

```python
>>> urlparse(urljoin(None, '<Map></Map>'))
ParseResult(scheme='', netloc='', path='<Map></Map>', params='', query='', fragment='')
```

`scheme` is still `''`, so it will still be considered a path.